### PR TITLE
Allow JavaScript classes to be included in API documentation

### PR DESCRIPTION
**Pull request by maxbrunsfeld**
This will allow documented classes to be converted from CoffeeScript to JS.

### CODE REVIEW

**Comment by maxbrunsfeld** [comment]
In this PR, I've converted `Color` to JS just to demonstrate that it works. Equivalent documentation appears to have been generated for the class:

```diff
1078,1079c1078,1080
<       "filename": "src/color.coffee",
<       "srcUrl": "https://github.com/atom/atom/blob/v1.11.0-dev/src/color.coffee#L7",
---
>       "superClass": null,
>       "filename": "src/color.js",
>       "srcUrl": "https://github.com/atom/atom/blob/v1.11.0-dev/src/color.js#L7",
1085c1086
<           "srcUrl": "https://github.com/atom/atom/blob/v1.11.0-dev/src/color.coffee#L15",
---
>           "srcUrl": "https://github.com/atom/atom/blob/v1.11.0-dev/src/color.js#L15",
1109c1110
<           "srcUrl": "https://github.com/atom/atom/blob/v1.11.0-dev/src/color.coffee#L57",
---
>           "srcUrl": "https://github.com/atom/atom/blob/v1.11.0-dev/src/color.js#L79",
1123c1124
<           "srcUrl": "https://github.com/atom/atom/blob/v1.11.0-dev/src/color.coffee#L61",
---
>           "srcUrl": "https://github.com/atom/atom/blob/v1.11.0-dev/src/color.js#L84",
1138,1139c1139,1140
<       "summary": "A simple color class returned from {Config::get} when the value\nat the key path is of type 'color'. ",
<       "description": "A simple color class returned from {Config::get} when the value\nat the key path is of type 'color'. "
---
>       "summary": "A simple color class returned from {Config::get} when the value\nat the key path is of type 'color'.",
>       "description": "A simple color class returned from {Config::get} when the value\nat the key path is of type 'color'."
```

**Comment by maxbrunsfeld** [comment]
I converted two more small files to JS and found another small bug in the new documentation gathering system, [joanna](https://github.com/maxbrunsfeld/joanna).

**Comment by maxbrunsfeld** [comment]
Ok, after testing this with 4 classes, I think it's good enough for now.

**Comment by nathansobo** [comment]
✨

File: src/window.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Public: Measure how long a function takes to run.
+//
+// description - A {String} description that will be logged to the console when
+//               the function completes.
+// fn - A {Function} to measure the duration of.
+//
+// Returns the value returned by the given function.
+window.measure = function (description, fn) {
+  let start = Date.now()
```
Comment (steelbrain) [inline src/window.js]: btw, there's also a `performance.now()`

Comment (steelbrain) [inline src/window.js]: or maybe this whole diffing homegrown function could be replaced by `console.time(description)` and `console.timeEnd(description)`

Comment (nathansobo) [inline src/window.js]: Sounds good if you want to open a focused PR.

**Pull request merged**